integration/integration_linux_test.go

```diff
@@ -20,64 +20,181 @@ import (
+			contNS1                                           Namespace
+			contNS2                                           Namespace
+			basicBridgeEnv                                    TestEnv
+			chainedBridgeBandwidthEnv                         TestEnv
+			chainedBridgeBandwidthSession, basicBridgeSession *gexec.Session
+			chainedBridgeBandwidthPort, basicBridgePort       int
+			chainedBridgeIP, basicBridgeIP                    string
+			runtimeWithLimit, runtimeWithoutLimit             time.Duration
+		)
 
-		addrOutput := env.runInNS(contNS, "ip", "addr")
-		Expect(addrOutput).To(ContainSubstring(expectedIPPrefix))
+		BeforeEach(func() {
+			hostNS = Namespace(fmt.Sprintf("cni-test-host-%x", rand.Int31()))
+			hostNS.Add()
 
-		env.runInNS(hostNS, cnitoolBin, "del", netName, contNS.LongName())
-	}
+			contNS1 = Namespace(fmt.Sprintf("cni-test-cont1-%x", rand.Int31()))
+			contNS1.Add()
 
-	It("supports basic network add and del operations", func() {
-		basicAssertion("basic-ptp", "10.1.2.")
-	})
+			contNS2 = Namespace(fmt.Sprintf("cni-test-cont2-%x", rand.Int31()))
+			contNS2.Add()
 
-	It("supports add and del with ptp + bandwidth", func() {
-		basicAssertion("chained-ptp-bandwidth", "10.9.2.")
-	})
+			basicBridgeNetConfPath, err := filepath.Abs("./testdata/basic-bridge")
+			Expect(err).NotTo(HaveOccurred())
+
+			basicBridgeEnv = TestEnv([]string{
+				"CNI_PATH=" + cniPath,
+				"NETCONFPATH=" + basicBridgeNetConfPath,
+				"PATH=" + os.Getenv("PATH"),
+			})
+
+			chainedBridgeBandwidthNetConfPath, err := filepath.Abs("./testdata/chained-bridge-bandwidth")
+			Expect(err).NotTo(HaveOccurred())
+
+			chainedBridgeBandwidthEnv = TestEnv([]string{
+				"CNI_PATH=" + cniPath,
+				"NETCONFPATH=" + chainedBridgeBandwidthNetConfPath,
+				"PATH=" + os.Getenv("PATH"),
+			})
+		})
+
+		AfterEach(func() {
+			if chainedBridgeBandwidthSession != nil {
+				chainedBridgeBandwidthSession.Kill()
+			}
+			if basicBridgeSession != nil {
+				basicBridgeSession.Kill()
+			}
+
+			chainedBridgeBandwidthEnv.runInNS(hostNS, cnitoolBin, "del", "network-chain-test", contNS1.LongName())
+			basicBridgeEnv.runInNS(hostNS, cnitoolBin, "del", "network-chain-test", contNS2.LongName())
+		})
+
+		Measure("limits traffic only on the restricted bandwith veth device", func(b Benchmarker) {
+			ipRegexp := regexp.MustCompile("10\\.11\\.2\\.\\d{1,3}")
+
+			By(fmt.Sprintf("adding %s to %s\n\n", "chained-bridge-bandwidth", contNS1.ShortName()), func() {
+				chainedBridgeBandwidthEnv.runInNS(hostNS, cnitoolBin, "add", "network-chain-test", contNS1.LongName())
+				chainedBridgeIP = ipRegexp.FindString(chainedBridgeBandwidthEnv.runInNS(contNS1, "ip", "addr"))
+				Expect(chainedBridgeIP).To(ContainSubstring("10.11.2."))
+			})
 
-	It("supports add and del with bridge + bandwidth", func() {
-		basicAssertion("chained-bridge-bandwidth", "10.11.2.")
+			By(fmt.Sprintf("adding %s to %s\n\n", "basic-bridge", contNS2.ShortName()), func() {
```

Comment:
nit: I slightly prefer the `By(...)` syntax where there isn't another nested `func()`, e.g. https://onsi.github.io/ginkgo/#documenting-complex-its-by

This makes it so you don't have to do weird things like lift variables outside.

But no big deal.

Reply:
We removed the nesting and localized some vars.